cmake/developer_package/packaging/rpm.cmake

```diff
@@ -0,0 +1,273 @@
+
+    # non-native stuff
+    set(OV_CPACK_PYTHONDIR ${OV_CPACK_PLUGINSDIR})
+    set(OV_CPACK_SHAREDIR ${CMAKE_INSTALL_DATADIR}/openvino-${OpenVINO_VERSION}) # internal
+    set(OV_CPACK_SAMPLESDIR ${OV_CPACK_SHAREDIR}/samples)
+    set(OV_CPACK_DEVREQDIR ${OV_CPACK_SHAREDIR})
+    unset(OV_CPACK_SHAREDIR)
+
+    # skipped during rpm packaging
+    set(OV_CPACK_WHEELSDIR "tools")
+
+    # for BW compatibility
+    set(IE_CPACK_LIBRARY_PATH ${OV_CPACK_LIBRARYDIR})
+    set(IE_CPACK_RUNTIME_PATH ${OV_CPACK_RUNTIMEDIR})
+    set(IE_CPACK_ARCHIVE_PATH ${OV_CPACK_ARCHIVEDIR})
+endmacro()
+
+ov_rpm_cpack_set_dirs()
+
+#
+# Override CPack components name for Debian generator
+# This is needed to change the granularity, i.e. merge several components
+# into a single one
+#
+
+macro(ov_override_component_names)
+    # merge C++ and C runtimes
+    set(OV_CPACK_COMP_CORE_C "${OV_CPACK_COMP_CORE}")
+    set(OV_CPACK_COMP_CORE_C_DEV "${OV_CPACK_COMP_CORE_DEV}")
+    # merge all pythons into a single component
+    set(OV_CPACK_COMP_PYTHON_OPENVINO "pyopenvino")
+    set(OV_CPACK_COMP_PYTHON_IE_API "${OV_CPACK_COMP_PYTHON_OPENVINO}")
+    set(OV_CPACK_COMP_PYTHON_NGRAPH "${OV_CPACK_COMP_PYTHON_OPENVINO}")
+    # merge all C / C++ samples as a single samples component
+    set(OV_CPACK_COMP_CPP_SAMPLES "samples")
+    set(OV_CPACK_COMP_C_SAMPLES "${OV_CPACK_COMP_CPP_SAMPLES}")
+    # set(OV_CPACK_COMP_PYTHON_SAMPLES "${OV_CPACK_COMP_CPP_SAMPLES}")
+    # move requirements.txt to core-dev
+    set(OV_CPACK_COMP_DEV_REQ_FILES "${OV_CPACK_COMP_CORE_DEV}")
+    # move core_tools to core-dev
+    set(OV_CPACK_COMP_CORE_TOOLS "${OV_CPACK_COMP_CORE_DEV}")
+endmacro()
+
+ov_override_component_names()
+
+#
+# Common Debian specific settings
+#
+
+macro(ov_rpm_specific_settings)
+    # multiple packages are generated
+    set(CPACK_RPM_COMPONENT_INSTALL ON)
+    # automatically find dependencies for binaries
+    set(CPACK_RPM_PACKAGE_SHLIBDEPS ON)
+    # enable dependencies between components
+    set(CPACK_RPM_ENABLE_COMPONENT_DEPENDS ON)
+    # control file permissions
+    set(CPACK_RPM_PACKAGE_CONTROL_STRICT_PERMISSION OFF)
+    # homepage
+    set(CPACK_RPM_PACKAGE_HOMEPAGE "https://docs.openvino.ai/")
+    # enable for debug cpack run
+    if(NOT DEFINED CPACK_RPM_PACKAGE_DEBUG)
+        set(CPACK_RPM_PACKAGE_DEBUG OFF)
+    endif()
+
+    # WA: dpkg-shlibdeps requires folder with libraries
+    # proper way is to use -l (path to libs) and -L (path to shlibs) for other already installed components
+    # but it requires CMake source code changes
+    # with current WA automatic deps detection via dpkg-shlibdeps for "our libraries"
+    # is ignored; but dependencies between our components are here because of
+    # CPACK_COMPONENT_<UCOMP>_DEPENDS variables
+    # More proper WA is try to enable INSTALL_RPATH
+
+    if(DEFINED CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+        set(CPACK_RPM_PACKAGE_SHLIBDEPS_PRIVATE_DIRS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
```

Comment:
no such var
need to understand how dependencies are searched in rpm case

Reply:
There is CPACK_RPM_PACKAGE_AUTOREQPROV for automatic listing of shared libraries and automatic shared libraries dependency detection at the same time.
https://cmake.org/cmake/help/v3.8/module/CPackRPM.html?highlight=cpack_rpm_package#variable:CPACK_RPM_PACKAGE_AUTOREQPROV